osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -32,8 +33,26 @@ public abstract class ModFlashlight : Mod
         public override ModType Type => ModType.DifficultyIncrease;
         public override string Description => "Restricted view area.";
 
+        [SettingSource("Change radius based on combo", "Decrease the flashlight radius as combo increases.")]
+        public Bindable<bool> ChangeRadius { get; private set; }
+
+        [SettingSource("Initial radius", "Initial radius of the flashlight area.")]
+        public BindableNumber<float> InitialRadius { get; private set; }
+
+        public abstract float DefaultRadius { get; }
+
+        public abstract bool DefaultComboDependency { get; }
```

Comment:
these are not required if `ChangeRadius` and `InitialRadius` are declared abstract to begin with, then you can arbitrarily set defaults in overriding classes

Reply:
also the "radius" setting should probably be a multiplier applied to the default (i.e. be in the range 0.5x-1.5x regardless of ruleset), rather than that being multiplied by some ruleset specific number

Reply:
ok thanks for this i wanted to override to set custom max sizes but i wasnt sure how to

Reply:
> these are not required if `ChangeRadius` and `InitialRadius` are declared abstract to begin with, then you can arbitrarily set defaults in overriding classes

mind elaborating on this a bit, i tried declaring them abstract and it wouldnt allow me to override them

Reply:
In `ModFlashlight`, declare only the following:

```csharp
[SettingSource("Change radius based on combo", "Decrease the flashlight radius as combo increases.")]
public abstract Bindable<bool> ChangeRadius { get; }

[SettingSource("Initial radius", "Initial radius of the flashlight area.")]
public abstract BindableNumber<float> InitialRadius { get; }
```

Note the `abstract` spec on both. Now derived classes are free to implement the initial value of both as they wish, including setting their own values of `MinValue`, `MaxValue`, `Default`, `Precision` et al.

Compare `ModTimeRamp` and its implementations - `ModWind{Up,Down}` for a pre-existing example of this.